mdagent/tools/base_tools/analysis_tools/salt_bridge_tool.py

```diff
@@ -0,0 +1,104 @@
+            donor_residues = traj.topology.select(f'residue_name == "{pair[0]}"')
+            acceptor_residues = traj.topology.select(f'residue_name == "{pair[1]}"')
+
+            for donor_idx in donor_residues:
+                for acceptor_idx in acceptor_residues:
+                    distances = md.compute_distances(traj, [[donor_idx, acceptor_idx]])
+                    if any(d <= threshold_distance for d in distances):
+                        salt_bridges.append((donor_idx, acceptor_idx))
+
+ # Check if the donor and acceptor form a salt bridge
+                        if any(d <= threshold_distance for d in distances):
+                            # If yes, remove them from the unpaired set
+                            self.unpaired_residues.discard(donor_idx)  # Remove donor from unpaired residues set
+                            self.unpaired_residues.discard(acceptor_idx)  # Remove acceptor from unpaired residues set
+                        else:
+                            # If not, add them to the unpaired set
+                            self.unpaired_residues.add(donor_idx)  # Add donor to unpaired residues set
+                            self.unpaired_residues.add(acceptor_idx)  # Add acceptor to unpaired residues set
+        print("Salt bridges found:")
+        for bridge in salt_bridges:
+            print(
+                f"Residue {traj.topology.atom(bridge[0]).residue.index + 1} ({traj.topology.atom(bridge[0]).residue.name}) - "
+                f"Residue {traj.topology.atom(bridge[1]).residue.index + 1} ({traj.topology.atom(bridge[1]).residue.name})"
+            )
+
+            #Print unpaired residues
+        print("Unpaired_residues:")
+        salt_bridge_function = salt_bridge_tool.salt_bridge_function
+        for residue_idx in salt_bridge_function.unpaired_residues:
+            print(f"Residue {traj.topology.atom(residue_idx).residue.index + 1} ({traj.topology.atom(residue_idx).residue.name})")
+
+        return salt_bridges, list(self.unpaired_residues), list(residue_pairs)
+
+
+
+class SaltBridgeTool(BaseTool):
+        name = "salt_bridge_tool"
+        description = "A tool to find salt bridge in a protein trajectory"
+
+     def __init__(self, path_registry):
+         self.salt_bridge_function = SaltBridgeFunction(path_registry)
+
+    def _run(self, traj_file, top_file, threshold_distance=0.4, residue_pairs=None):
+         # Load trajectory using MDTraj
+        traj = md.load(traj_file, top=top_file)
+        #calls the salt bridge function
+        salt_bridges = [self.salt_bridge_function.find_salt_bridges(traj_file, top_file,
+        threshold_distance, residue_pairs)]
+        return salt_bridges
+
+    def _agg_result(self, result):
+        return result
+
+    def _call__(self, traj_file, top_file, threshold_distance=0.4, residue_pairs=None):
+        result = self._run(traj_file, top_file, threshold_distance, residue_pairs)
+        return self._agg_result(result)
```

Comment:
these two functions `_agg_result` and `_call__`  aren't needed for LangChain BaseTool.